dspace-api/src/main/java/org/dspace/app/statistics/ListNumberOfDownloadedBitstreamsPerInstitution.java

```diff
@@ -0,0 +1,168 @@
+            List<Community> topCommunities = communityService.findAllTop(context);
+            for (Community topCommunity : topCommunities) {
+                String communityID = topCommunity.getID().toString();
+                String numFound = new ListNumberOfDownloadedBitstreamsPerInstitution()
+                        .findDownloadedItems(solrServer + "/statistics/select?", communityID, year);
+                instCountMap.put(topCommunity.getName(), numFound);
+            }
+
+            System.out.println("\n\n");
+            printResult(instCountMap, year);
+
+            // Close database connection - no updates
+            context.abort();
+
+        } catch (SQLException | ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void printResult(Map<String, String> instCountMap, String year) {
+        System.out.println("Statistikk for nedlastinger " + year + ":\n\n");
+        for (String inst : new TreeSet<String>(instCountMap.keySet())) {
+            String count = instCountMap.get(inst);
+            String s = inst + ": " + count;
+            System.out.println(s);
+        }
+    }
+
+    public String findDownloadedItems(String solrHost, String topCommunityId, String year) {
+        String solrQuery = generateQueryBitStream(topCommunityId, year);
+        return requestSolr(solrHost, solrQuery);
+    }
+
+    private String generateQueryBitStream(String topCommunityId, String year) {
+        StringBuilder solrQuery = new StringBuilder();
+        solrQuery.append("q=type%3A0"); // org.dspace.core.Constants.BITSTREAM = 0;
+        solrQuery.append("&");
+        solrQuery.append("fq=-%28bundleName%3A%5B*+TO+*%5D-bundleName%3AORIGINAL%29");
+        solrQuery.append("&");
+        solrQuery.append("fq=owningItem%3A%5B*+TO+*%5D");
+        solrQuery.append("&");
+        solrQuery.append("fq=owningComm%3A");
+        solrQuery.append(topCommunityId);
+        solrQuery.append("&");
+        solrQuery.append("fq=owningColl%3A%5B*+TO+*%5D");
+        solrQuery.append("&");
+        solrQuery.append("fq=time%3A");
+        solrQuery.append(year);
+        solrQuery.append("*");
+        solrQuery.append("&");
+        solrQuery.append("fq=-isBot%3Atrue");
+        solrQuery.append("&");
+        solrQuery.append("rows=10");
+        return solrQuery.toString();
+    }
+
+    private String requestSolr(String solrHost, String solrQuery) {
+        String uri = solrHost + solrQuery;
+        //System.out.println("Using uri: " + uri);
+        String numFound = "";
+        HttpClient client = new HttpClient();
+        GetMethod method = new GetMethod(uri);
+        try {
+            String responseBody = "";
+            int statusCode = client.executeMethod(method);
+            if (statusCode == HttpStatus.SC_OK) {
+                BufferedReader rd = new BufferedReader(new InputStreamReader(
```

Comment:
**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *rd*. The resource is referenced in statements at the following lines: 133, 137. The resource closure statement is at line: 140. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by BufferedReader.readLine. Either a) close *rd* in a try-finally block or b) close the resource by declaring *rd* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).